_posts/2018-11-29-GraphQL-Stitching.markdown

```diff
@@ -0,0 +1,386 @@
+---
+layout: epic
+title: GraphQL Stitching 101
+date: 2018-11-29
+author: [orta]
+categories: [graphql, stitching, metaphysics]
+css: graphql-stitch
+comment_id: 450
+---
+
+Micro-Services make sense. You can scope a domain of your business to particular small unit of abstraction like an
+API. Doing so makes it easy to work in isolation, experiment with new ideas and evolve in many directions.
+
+We've been [carefully pushing][monolith] for years to move away from our single monolithic API, to a collection of
+smaller, more focused projects. Our [highlights docs][high] showcase this well. The movement to smaller composable
+services works great from an isolated platform/systems perspective but can be a bit tricky to handle with front-end
+clients. Until 2018, the way that we've addressed the growing complexity in our service later has been to migrate
+the complexity inside our main GraphQL API, [metaphysics][mp]. Metaphysics is our GraphQL API gateway that
+consolidates many API sources into a single service, then extending and interleaving and their data to make clients
+easier to write.
+
+However, as more services have been created, and grown - so has metaphysics. This creates a worrying trend, as the
+growth of code in metaphysics isn't quite linear.
+
+Our main line-of-thought on how to address this is via GraphQL schema stitching. We've been [running
+experiments][ex] in stitching for over a year, and have have been running with stitching enabled in production for
+a few months.
+
+<!-- more -->
+
+## What is Schema Stitching?
+
+The core idea behind schema stitching is that because GraphQL talks in type systems, you should be able to merge
+type systems from many GraphQL APIs into a single source of truth. Schema Stitching came out at the [end of
```

Comment:
`Schema Stitching` vs `schema stitching` in the previous sentence. Either capitalization works.